js/charts/clusteredverticalbar.js

```diff
@@ -0,0 +1,1092 @@
+        _measureDisplayName = [],
+        _measureFontStyle = [],
+        _measureFontWeight = [],
+        _measureFontSize = [],
+        _measureNumberFormat = [],
+        _measureTextColor = [],
+        _measureDisplayColor = [],
+        _measureBorderColor = [],
+        _tooltip;
+
+    /* These are the common variables that is shared across the different private/public 
+     * methods but is initialized/updated within the methods itself.
+     */
+    var _localSVG,
+        _localTotal = [],
+        _localXAxis,
+        _localYAxis,
+        _localXGrid,
+        _localYGrid,
+        _localData,
+        _localXLabels = [],
+        _localMin = 0,
+        _localMax = 0,
+        _localLegend,
+        _localTooltip;
+        // _localLabelStack;
+
+    /* These are the common private functions that is shared across the different private/public 
+     * methods but is initialized beforehand.
+     */
+    var _xDimension = d3.scaleBand(),
+        _xDimensionGrid = d3.scaleLinear(),
+        _xMeasure = d3.scaleBand(),
+        _y = d3.scaleLinear();
+
+    /* -------------------------------------------------------------------------------- */
+    var _setConfigParams = function(config) {
+        this.dimension(config.dimension);
+        this.measure(config.measure);
+        this.xAxis(config.xAxis);
+        this.yAxis(config.yAxis);
+        this.xAxisColor(config.xAxisColor);
+        this.yAxisColor(config.yAxisColor);
+        this.xAxisLabel(config.xAxisLabel);
+        this.yAxisLabel(config.yAxisLabel);
+        this.legend(config.legend);
+        this.legendPosition(config.legendPosition);
+        this.grid(config.grid);
+        this.dimensionDisplayName(config.dimensionDisplayName);
+        this.measureShowValue(config.measureShowValue);
+        this.measureDisplayName(config.measureDisplayName);
+        this.measureFontStyle(config.measureFontStyle);
+        this.measureFontWeight(config.measureFontWeight);
+        this.measureFontSize(config.measureFontSize);
+        this.measureNumberFormat(config.measureNumberFormat);
+        this.measureTextColor(config.measureTextColor);
+        this.measureDisplayColor(config.measureDisplayColor);
+        this.measureBorderColor(config.measureBorderColor);
+        this.tooltip(config.tooltip);
+    }
+
+    var _setAxisColor = function(axis, color) {
+        var path = axis.select('path'),
+            ticks = axis.selectAll('.tick');
+
+        path.style('stroke', color);
+
+        ticks.select('line')
+            .style('stroke', color);
+
+        ticks.select('text')
+            .style('fill', color);
+    }
+
+    /* Builds the html data for the tooltip
+     *
+     * @param {object} datum Datum forming the bar
+     * @param {function} chart Clustered Vertical Bar chart function
+     * @return {string} String encoded HTML data
+     */
+    var _buildTooltipData = function(datum, chart) {
+        var output = "";
+
+        output += "<table><tr>"
```

Comment:
Agreed @techkpi, but in this case, the reason I didn't do it was to make the code more readable since there are multiple variables in use and it's in an HTML structure.

Reply:
ok, thats fine @suvrat-joshi . I am going ahead and merging your PR. Also, once merged, please access how Khushbu's PR would be affected